File: app/src/main/java/com/example/plotting_fe/mypage/ui/MypageProfileFragment.java
package com.example.plotting_fe.mypage.ui;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.example.plotting_fe.R;
import com.example.plotting_fe.utils.Utils;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link MypageProfileFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class MypageProfileFragment extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    public MypageProfileFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment MpageProfileFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static MypageProfileFragment newInstance(String param1, String param2) {
        MypageProfileFragment fragment = new MypageProfileFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_mypage_profile, container, false);

        // 뒤로 가기 버튼 설정
        ImageView backButton = view.findViewById(R.id.iv_back);
        Utils.onBackButtonClick(getActivity(), backButton);

        // 수정 버튼 클릭 리스너 설정
        ImageView editButton = view.findViewById(R.id.iv_edit);
        editButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d("ProfileActivity", "Edit button clicked"); // 로그 추가
                // SampleActivity로 전환하는 Intent 생성
                Intent intent = new Intent(getActivity(), EditProfileActivity.class);
                startActivity(intent); // 액티비티 시작
            }
        });

        // Inflate the layout for this fragment
        return view;
    }
}

File: app/src/main/java/com/example/plotting_fe/mypage/ui/MypageProfileFragment.kt
package com.example.plotting_fe.mypage.ui

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.widget.AppCompatImageView
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.example.plotting_fe.R
import com.example.plotting_fe.global.ResponseTemplate
import com.example.plotting_fe.global.util.ApiClient
import com.example.plotting_fe.global.util.Utils
import com.example.plotting_fe.mypage.dto.response.MyProfileResponse
import com.example.plotting_fe.mypage.presentation.MypageController
import com.example.plotting_fe.myplogging.presentation.MyPloggingController
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MypageProfileFragment: Fragment() {
    private lateinit var profileImageView : AppCompatImageView
    private lateinit var nicknameTextView : TextView
    private lateinit var emailTextView : TextView
    private lateinit var profileMessageTextView : TextView
    private lateinit var publicTextView : TextView

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_mypage_profile, container, false)

        // UI 요소 초기화
        profileImageView = view.findViewById(R.id.iv_image)
        nicknameTextView = view.findViewById(R.id.tv_nickname_value)
        emailTextView = view.findViewById(R.id.tv_email_value)
        profileMessageTextView = view.findViewById(R.id.tv_intro_value)
        publicTextView = view.findViewById(R.id.visibility_spinner)

        // 뒤로 가기 버튼 설정
        val backButton: ImageView = view.findViewById(R.id.iv_back)
        Utils.onBackButtonClick(this, backButton)

        // 수정 버튼 클릭 리스너 설정
        val editButton: ImageView = view.findViewById(R.id.iv_edit)
        editButton.setOnClickListener {
            Log.d("ProfileActivity", "Edit button clicked")
            // SampleActivity로 전환하는 Intent 생성
            val intent = Intent(activity, EditProfileActivity::class.java)
            startActivity(intent) // 액티비티 시작
        }

        loadProfile()

        return view
    }

    private fun loadProfile(){
        val myPageController = ApiClient.getApiClient().create(MypageController::class.java)
        myPageController.getMyProfile().enqueue(object : Callback<ResponseTemplate<MyProfileResponse>> {
            override fun onResponse(
                call: Call<ResponseTemplate<MyProfileResponse>>,
                response: Response<ResponseTemplate<MyProfileResponse>>
            ) {
                if(response.isSuccessful){
                    Log.d("get", "onResponse 성공: " + response.body().toString())
                    val profile = response.body()?.results
                    if(profile!=null){
                        updateUI(profile)
                    }else{
                        Log.d("get", "onResponse 성공: profile == null")
                    }
                }else{
                    Log.d("get", "onResponse 실패: " + response.code())
                }
            }
            override fun onFailure(call: Call<ResponseTemplate<MyProfileResponse>>, t: Throwable) {
                Log.d("get", "onFailure 에러: " +  t.message.toString())

            }
        })
    }

    private fun updateUI(profile: MyProfileResponse){
        nicknameTextView.text = profile.nickname
        emailTextView.text = profile.email
        profileMessageTextView.text = profile.profileMessage
        Glide.with(this)
            .load(profile.profileImageUrl)
            .placeholder(R.drawable.ic_flower)
            .error(R.drawable.ic_flower)
            .into(profileImageView)
        publicTextView.text = if(profile.isProfileResponse) "공개" else "비공개"
    }

}